### PR TITLE
WFLY-9664 Add support for scattered cache mode

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.jboss.org/browse/WFLY-9664

Design doc: https://github.com/wildfly/wildfly-proposals/pull/12

### CODE REVIEW

**Comment by kabir** [comment]
@pferraro adding the 'hold' label until @mvinkler has done Test Dev

File: docs/src/main/asciidoc/_high-availability/subsystem-support/Infinispan_Subsystem.adoc [hunk 1]
```diff
@@ -112,11 +117,34 @@ To create a distributed cache where a given entry is stored on 3 nodes:
...
+This design means that a scattered cache only requires 1 remote invocation to write a given cache entry, regardless of which node initiated the cache operation.
+By comparison, a distributed cache (with 2 owners) uses 1 remote invocation to write a cache entry if and only if the primary owner initiated the cache operation, and otherwise requires 2 remote invocations.
+
+For more information about this cache type, refer to the http://infinispan.org/docs/stable/user_guide/user_guide.html#scattered_mode[the Infinispan documentation].
+
+To create a scattered cache:
+
+[source]
+----
+/subsystem=infinispan/cache-container=foo/distributed-cache=bar:add()
```
Comment (mvinkler) [inline docs/src/main/asciidoc/_high-availability/subsystem-support/Infinispan_Subsystem.adoc]: This command is not valid.

**Comment by kabir** [comment]
Retest this please

File: clustering/infinispan/extension/src/main/resources/schema/jboss-as-infinispan_6_0.xsd [hunk 2]
```diff
@@ -0,0 +1,1099 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!--
+  ~ JBoss, Home of Professional Open Source.
+<<<<<<< HEAD
+  ~ Copyright 2018, Red Hat, Inc., and individual contributors
+=======
+  ~ Copyright 2011, Red Hat, Inc., and individual contributors
+>>>>>>> WFLY-9664 Add support for scattered cache.
```
Comment (kabir) [inline clustering/infinispan/extension/src/main/resources/schema/jboss-as-infinispan_6_0.xsd]: @pferraro rebase error

Comment (pferraro) [reply clustering/infinispan/extension/src/main/resources/schema/jboss-as-infinispan_6_0.xsd]: oops.

Comment (pferraro) [reply clustering/infinispan/extension/src/main/resources/schema/jboss-as-infinispan_6_0.xsd]: Fixed.

**Pull request merged**